### ISSUE TITLE
Game.ResetElapsedTime() throws NullReferenceException

### ISSUE THREAD

**Issue by nkast** [MonoGame/MonoGame#6112]
I was trying to port an old XNA sample project where `ResetElapsedTime()` was called from LoadContent().
The cause is `Game._gameTimer` which in MonoGame isn't initialized until the first Tick/Update.

MonoGame 3.7.1041.
The tested platform was Desktop/DX but the bug is in Game class and should affect all platforms.

**Comment by Jjagg** [comment]
Nice catch!

**Comment by nkast** [comment]
The fix to this is simple , all we have to do is pre-initialize Game._gameTimer 
`private Stopwatch _gameTimer = new Stopwatch();`
However that revealed a couple of bugs in Tick() and Tests.

Some test cases of `test GameTest.Disposal.Method_does_not_throw_after_Dispose()`
will throw NullReferenceException but the test  
checks only for `AssertDoesNotThrow<ObjectDisposedException>()` and pass.

```
=> Method_does_not_throw_after_Dispose("ResetElapsedTime")
AssertDoesNotThrow<System.ObjectDisposedException> caught and ignored System.NullReferenceException: Object reference not set to an instance of an object.
   at MonoGame.Tests.GameTest.Disposal.RunAndUnpackException(Action action) in P:\MonoGame\Test\Framework\GameTest.cs:line 94
   at MonoGame.Tests.GameTest.Disposal.<>c__DisplayClass1_0.<Method_does_not_throw_after_Dispose>b__0() in P:\MonoGame\Test\Framework\GameTest.cs:line 79
   at MonoGame.Tests.GameTest.Disposal.AssertDoesNotThrow[T](TestDelegate code) in P:\MonoGame\Test\Framework\GameTest.cs:line 124
=> Method_does_not_throw_after_Dispose("Run")
AssertDoesNotThrow<System.ObjectDisposedException> caught and ignored System.NullReferenceException: Object reference not set to an instance of an object.
   at MonoGame.Tests.GameTest.Disposal.RunAndUnpackException(Action action) in P:\MonoGame\Test\Framework\GameTest.cs:line 94
   at MonoGame.Tests.GameTest.Disposal.<>c__DisplayClass1_0.<Method_does_not_throw_after_Dispose>b__0() in P:\MonoGame\Test\Framework\GameTest.cs:line 79
   at MonoGame.Tests.GameTest.Disposal.AssertDoesNotThrow[T](TestDelegate code) in P:\MonoGame\Test\Framework\GameTest.cs:line 124
=> Method_does_not_throw_after_Dispose("Tick")
AssertDoesNotThrow<System.ObjectDisposedException> caught and ignored System.NullReferenceException: Object reference not set to an instance of an object.
   at MonoGame.Tests.GameTest.Disposal.RunAndUnpackException(Action action) in P:\MonoGame\Test\Framework\GameTest.cs:line 94
   at MonoGame.Tests.GameTest.Disposal.<>c__DisplayClass1_0.<Method_does_not_throw_after_Dispose>b__0() in P:\MonoGame\Test\Framework\GameTest.cs:line 79
   at MonoGame.Tests.GameTest.Disposal.AssertDoesNotThrow[T](TestDelegate code) in P:\MonoGame\Test\Framework\GameTest.cs:line 124
```

ResetElapsedTime() fails because MockGame.Platform = null.
Tick() fails because _gameTimer = null.

If I replace the initialization of _gameTimer with 
`private Stopwatch _gameTimer = new Stopwatch();`
the Tick method get's in an endless loop  and the Test halted indefinitely.


In conclusion:'
1) There's another way to fix this bug in ResetElapsedTime() with a check for null
```
if(_gameTimer != null)
{
  _gameTimer.Reset();
  _gameTimer.Start();
}
```
2) Remove the AssertDoesNotThrow<ObjectDisposedException> and match the behavior of XNA. 
If XNA throws an Exception then MG should throw that same Exception, otherwise it shouldn't fail.

3) See if there is anything we need to change/fic in Tick() to avoid an endless loop.

**Comment by nkast** [comment]
Here's another related bug.
in Program.cs replace Main() with this code

```
static void Main(string[] args)
        {
            using (Game1 game = new Game1())
            {
                //game.Run();
                game.Tick();
                game.Tick();
                game.Tick();
            }
        }
```
and remove GraphicsDevice.Clear() from Draw().

XNA will call Update()/Draw() and update GameTime. 
Will not call Initialize()/LoadContent().

MG will throw a Null exception in the first tick.

**Comment by corey-buchanan** [comment]
Looks like the first fixes have been committed. I'd like to take a stab at this last one, but I can't see to find the Program.cs file. Which namespace should this be in?
